test/unit/int_algolia/scripts/hooks/order/algoliaHooks.test.js

```diff
@@ -0,0 +1,214 @@
+        return this.productLineItems;
+    }
+}
+
+// Mock for dw.order.Order
+class OrderMock {
+    constructor({ orderNo, shipments } = {}) {
+        this.orderNo = orderNo || '12345';
+        this.shipments = shipments || [];
+    }
+
+    getShipments() {
+        return this.shipments;
+    }
+}
+
+describe('Algolia Hooks', function () {
+    // Mock these modules for tests
+    beforeAll(() => {
+        jest.mock('*/cartridge/scripts/algolia/lib/algoliaData', () => ({
+            getPreference: jest.fn().mockImplementation((id) => {
+                switch (id) {
+                    case 'InStockThreshold':
+                        return 5;
+                    case 'RecordModel':
+                        return 'master-level';
+                    case 'IndexOutOfStock':
+                        return true; 
+                    default:
+                        return null;
+                }
+            }),
+            getSetOfArray: jest.fn().mockImplementation((id) => {
+                switch (id) {
+                    case 'AdditionalAttributes':
+                        return ['storeAvailability', 'in_stock'];
+                    default:
+                        return [];
+                }
+            })
+        }), { virtual: true });
+    });
+
+    let mockMasterProduct;
+    let mockVariant, mockVariant2;
+    let mockOrder;
+    let mockShipmentInStore, mockShipmentStandard;
+
+    beforeEach(() => {
+        // Setup common test data
+        mockMasterProduct = new MasterVariantMock({
+            inventoryList: {
+                getRecord: jest.fn().mockReturnValue({ ATS: { value: 3 } })
+            }
+        });
+        mockVariant = new VariantMock({
+            ID: '701644031206M',
+            variationAttributes: { color: 'JJB52A0', size: '004' },
+            masterProduct: mockMasterProduct,
+        });
+
+        mockVariant2 = new VariantMock({
+            ID: '701644031207M',
+            variationAttributes: { color: 'JJB52A0', size: '004' },
+            masterProduct: mockMasterProduct,
+        });
+        
+        mockMasterProduct.variants = collectionHelper.createCollection([mockVariant, mockVariant2]);
+        
+        mockShipmentInStore = new ShipmentMock({
+            shipmentType: 'instore',
+            fromStoreId: 'store1',
+            productLineItems: [{
+                product: mockVariant
+            }]
+        });
+
+        mockShipmentStandard = new ShipmentMock({
+            shipmentType: 'standard',
+            productLineItems: [{
+                product: mockVariant2
+            }]
+        });
+
+        mockOrder = new OrderMock({
+            orderNo: '12345',
+            shipments: [mockShipmentInStore, mockShipmentStandard]
+        });
+    });
+
+    test('inventoryUpdate should handle in-store pickup shipment correctly', function () {
+        const Status = require('dw/system/Status');
+        // Arrange
+        const order = mockOrder;
```

Comment:
The test description doesn't reflects what's tested: the `mockOrder` also contains a standard shipments.
And the `inventoryUpdate` function always returns a `Status` object, even in case of error, so this test doesn't look very useful as-is

Reply:
You are right; it actually doesn't make sense because we are using try-catch. I also rehandled algoliaHooks tests to make it more comprehensive.